enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,290 @@
+running those pods, ingress traffic can no longer work. This is until DNS is
+adjusted (solving the problem temporarily) or a load-balancer / some virtual IP
+solution is put in place.
+
+We would like to avoid this complication.
+
+### Goals
+
+- Enable expansion of Single Node clusters by ensuring ingress traffic can always
+  be directed at the single control-plane node successfully, even when new workers
+  are added.
+
+### Non-Goals
+
+- Deal with expansion of the single-node control-plane by adding more control-plane nodes
+- Deal with expansion of clusters that have been installed before the implementation
+  of this enhancement (they can be take ncare of with a simple documentation on how it can be
+  enabled with a few manual steps).
+- Deal with "day 1" UPI installation of single-node control plane clusters
+  that have additional non-control plane worker nodes
+- Deal with cloud IPI installations of Single Node OpenShift. They do not suffer
+  suffer from the floating ingress issue as they're using a cloud provided load
+  balancer which is automatically configured (using a LoadBalancer type Service)
+  to follow the nodes currently holding the router-default pods.
+- Adjust Assisted Installer to use the baremetal platform for single-node installations.
+  The baremetal platforms solves this issue with virtual IP addresses/keepalived.
+- Deal with users who want their ingress traffic to not go through the single
+  control-plane node.
+
+## Proposal
+
+Modify the installer's behavior during Single Node "none"-platform installation
+to create an `IngressController` CR installer manifest targetting control-plane
+nodes rather than worker nodes, which will in turn prevent the OpenShift
+Ingress Operator from creating the default `IngressController` CR which targets
+worker nodes. This will ensure that the `router-default` deployment created by
+the OpenShift Cluster Ingress Operator will always run on the single control
+plane node, and as a result any `*.apps.<cluster>.<base>` DNS entries which
+originally pointed at the single control plane node will remain correct even in
+the face of newly added worker nodes.
+
+This is made possible due to [this](https://github.com/openshift/enhancements/blob/4938b44dc0373a032cae9a48dbe0f86e06f8a189/enhancements/ingress/user-defined-default-ingress-controller.md) previous enhancement.
+
+
+### User Stories
+
+- As an OpenShift Single Node administrator, I want to add worker nodes to my
+  single-node cluster, so that it'll be able to handle growing computation demands.
+
+### API Extensions
+
+This enhancement does not modify/add any API
+
+### Implementation Details/Notes/Constraints
+
+This enhancement can be easily implemented by adjusting the installer's
+`generateDefaultIngressController` method such that when `config.platform.none
+!= nil`, it will return an `IngressController` CR struct with
+`.spec.nodePlacement.nodeSelector.matchLabels` targetting
+`node-role.kubernetes.io/master`. 
```

Comment:
Is there a way to also detect that it's single node, at the moment I think this would also affect HA clusters that are platform none wouldn't it?

Reply:
You're right, introduced this error during my last enhancement commit. Will fix